pkg/dns/dns.go

```diff
@@ -153,7 +167,9 @@ func (kd *KubeDNS) updateConfig(nextConfig *config.Config) {
 				nameServers = append(nameServers, net.JoinHostPort(ip, port))
 			}
 		}
-		kd.SkyDNSConfig.Nameservers = nameServers
+		if len(nameServers) == 0 {
+			kd.SkyDNSConfig.Nameservers = kd.loadDefaultNameserver()
+		}
```

Comment:
Seems like `kd.SkyDNSConfig.Nameservers = nameServers` is still needed?

Reply:
Fixed..